### PR TITLE
pyslurm: bump to unbreak

### PR DESCRIPTION
**Pull request by veprbl**
Fixes: bd9512f217 ('slurm: 17.02.6 -> 17.02.9 for CVE-2017-15566')

###### Motivation for this change

Update is needed to make pyslurm work with never slurm.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/pyslurm/default.nix [hunk 1]
```diff
@@ -6,12 +6,10 @@ buildPythonPackage rec {
   src = fetchFromGitHub {
     repo = "pyslurm";
     owner = "PySlurm";
-    rev = "69e4f4fd66003b98ddb7da25613fe641d4ae160d";
-    sha256 = "051kafkndbniklxyf0drb360aiblnqcf9rqjbvmqh66zrfya1m28";
+    rev = "a2acbc820da419e308c5817998d2abe78a7b75e6";
```
Comment (andir) [inline pkgs/development/python-modules/pyslurm/default.nix]: There was a version bump in the upstream repo just yesterday (https://github.com/PySlurm/pyslurm/commit/f717492f8e91599ff413a6789872f43c60fe8d6d). Can't you use that?

Also adding the version to the `name` attribute would probably be good.

Comment (veprbl) [reply pkgs/development/python-modules/pyslurm/default.nix]: @andir No I can’t. The version I chose matching slurm 17.02. The latest pyslurm will require slurm 17.11

Also note that this targets stable.

Comment (andir) [reply pkgs/development/python-modules/pyslurm/default.nix]: ok, makes sense.

**Comment by andir** [comment]
@GrahamcOfBorg build pyslurm

**Comment by andir** [comment]
@veprbl Are you able to get the package tests working? It seems like you have to add `nose` to `checkInputs` and define `checkPhase = "cd ..; nose pyslurm-${src.rev}-src/tests"`but then the tests fail. I haven't used `pyslurm` or `slurm` before so maybe you have an idea.

We will also need this patch or a bump (of both packges) on master.

**Comment by veprbl** [comment]
@andir I don't think you can test this in checkPhase as it requires a running server.

**Comment by veprbl** [comment]
@GrahamcOfBorg build python3Packages.pyslurm python2Packages.pyslurm

**Comment by veprbl** [comment]
The fix for master: #33656

**Comment by veprbl** [comment]
@andir Thanks!

**Pull request merged**